### DISCUSSION TITLE
PostMessage() - Disable retry on rate limit

### DISCUSSION THREAD

**Discussion by yourfavorite** [slack-go/slack#1023] [Q&A] [answered]
Is there any way to disable the PostMessage() retry after rate limit? Thanks!

**Comment by kanata2** [comment, marked as answer]
You can check whether the returned error is due to a rate limit or not.
So how about switching the handling way based on this result?

**Reply by yourfavorite** [reply]
How would this work? Right now I don't tell the package to retry after a failure. I simply log the error and expect to be done with it. How can I tell the package not to retry?

```
	_, _, err := client.PostMessage(c, slack.MsgOptionText(m, false))
	if err != nil {
		return err
	}
```

**Reply by yourfavorite** [reply]
I'm sorry. I misread the error message. Had a bunch of messages post at the same time and assumed that the library was retrying when in reality there were actually a ton of messages posting in a short period of time. Instead the package is just returning an error telling me to retry: `slack rate limit exceeded, retry after 1s`. This explains why I couldn't find any retry logic in the package for this function 😄  Thanks!
